src/Infrastructure/BackgroundJobs/RecurringJobs/RecurringJobInitialization.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Finbuckle.MultiTenant;
+using FSH.WebApi.Application.Catalog.Products.RecurringJobs;
+using FSH.WebApi.Application.Common.Interfaces;
+using FSH.WebApi.Infrastructure.Multitenancy;
+using Hangfire;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace FSH.WebApi.Infrastructure.BackgroundJobs.RecurringJobs;
+
+public class RecurringJobInitialization : IRecurringJobInitialization
+{
+    private readonly ILogger<RecurringJobInitialization> _logger;
+    private readonly IJobService _jobService;
+    private readonly TenantDbContext _tenantDbContext;
+    private readonly IServiceProvider _serviceProvider;
+    private readonly ITenantInfo _tenantInfo;
+
+    public RecurringJobInitialization(ILogger<RecurringJobInitialization> logger, IJobService jobService, TenantDbContext tenantDbContext, IServiceProvider serviceProvider, ITenantInfo tenantInfo)
+    {
+        _logger = logger;
+        _jobService = jobService;
+        _tenantDbContext = tenantDbContext;
+        _serviceProvider = serviceProvider;
+        _tenantInfo = tenantInfo;
+    }
+
+    public async Task InitializeJobsForTenantAsync(CancellationToken cancellationToken)
+    {
+        foreach (var tenant in await _tenantDbContext.TenantInfo.ToListAsync(cancellationToken))
+        {
+            InitializeJobsForTenant(tenant);
+        }
+    }
+
+    public void InitializeJobsForTenant(FSHTenantInfo tenant)
+    {
+        // First create a new scope
+        using var scope = _serviceProvider.CreateScope();
+
+        // Then set current tenant so the right connectionstring is used
+        scope.ServiceProvider.GetRequiredService<IMultiTenantContextAccessor>()
+            .MultiTenantContext = new MultiTenantContext<FSHTenantInfo>()
+            {
+                TenantInfo = tenant
+            };
+
+        scope.ServiceProvider.GetRequiredService<IRecurringJobInitialization>()
+            .InitializeRecurringJobs();
+    }
+
+    public void InitializeRecurringJobs()
+    {
+        _jobService.AddOrUpdate<ICheckProductOrdersJob>("Id", x => x.CheckInProductOrders(), () => Cron.Minutely(), TimeZoneInfo.Utc, "default");
```

Comment:
At this point only a recurring job is created since it has the same name, one option is to set the tenat key as part of the name and this in turn is used when trying to initialize the job.

Reply:
![image](https://user-images.githubusercontent.com/22311691/153784861-9f86535e-0410-46d9-9ba9-13353fd0f8d3.png)

Reply:
The Id of the job is indeed an option... but I think it would be nicer to send them as actual parameters of the method.

Reply:
![image](https://user-images.githubusercontent.com/22311691/153785922-b707bd0c-f29e-46e3-a825-8d9bbf5df26e.png)
hangfire in its own "AddOrUpdate " declaration does not support parameters, the idea is that from the invocation of the task in the jobactivator you can define the connection strings, tenantid... in order to take advantage of the clean dependency injection inside of the class or method of the job since the process of creation and invocation is carried out from the activator

Reply:
yes it does... the arguments of the method also get serialized into the jobdefinition... check my code below... that's production code that's running for over 5 years already.

Reply:
![image](https://user-images.githubusercontent.com/775707/153786384-f498d3fd-c693-48e7-95eb-b81b5b5a801d.png)